builtin/mainmenu/dlg_config_world.lua

```diff
@@ -24,38 +24,72 @@ local function modname_valid(name)
 end
 
 local function get_formspec(data)
-	local mod = data.list:get_list()[data.selected_mod]
+	local mod = data.list:get_list()[data.selected_mod] or {name=""}
 
 	local retval =
 		"size[11.5,7.5,true]" ..
 		"label[0.5,0;" .. fgettext("World:") .. "]" ..
 		"label[1.75,0;" .. data.worldspec.name .. "]"
 
-	if mod == nil then
-		mod = {name=""}
-	end
-
 	local hard_deps, soft_deps = modmgr.get_dependencies(mod.path)
 
+	if mod.is_modpack then
+		local info = modmgr.get_modpack_description(mod.path)
+		if info == "" then
+			info = fgettext("No modpack description provided.")
+		end
+		retval = retval ..
+			"label[0,0.7;" .. info .. "]"
+	else
+		retval = retval ..
+			"label[0,0.7;" .. fgettext("Mod:") .. "]" ..
+			"label[0.75,0.7;" .. mod.name .. "]"
+
+		if hard_deps == "" then
```

Comment:
I did this change because some mods have lots of optional but no hard dependency (and the other way round), so the textlist got a scrollbar and became inconvenient.